Add task status callback

jp-dbt opened the pull request:
## What is this PR?
This is a:
- [ ] documentation update
- [ ] bug fix with no breaking changes
- [x] new functionality
- [ ] a breaking change

All pull requests from community contributors should target the `main` branch (default).

## Description & motivation
- Add a new method to update task status and remove the individual status update methods. This allows us to generically call it upstream
- Accept a callback url as part of the async endpoint request
- Every time task status is updated, send a post request to the callback url with the status. The call will be retried a max of 5 times
- Return the state ID as part of the async endpoint response. This is needed in workspace controller to find the working directory
- Class rename

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [ ] Spark
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models
- [ ] I have added an entry to CHANGELOG.md

racheldaniel commented on dbt_server/views.py:
```diff
@@ -195,7 +196,7 @@ def parse_project(args: ParseArgs):
 
 @app.post("/async/dbt", response_model=schemas.Task)
```
```suggestion
@app.post("/async/dbt")
```
Removing the `response_model` as this PR changes the response structure

racheldaniel reviewed (changes requested):
One change request, one nitpicky and up for negotiation request, and I think a changelog entry would be good for this one, but otherwise looks great!

jp-dbt commented on dbt_server/views.py:
```diff
@@ -216,12 +217,22 @@ async def dbt_entry_async(
 
 @app.post("/sync/dbt")
-async def dbt_entry_sync(args: dbtCommandArgs):
+async def dbt_entry_sync(args: DBTCommandArgs):
```
Sure, definitely on board with being consistent with the other Python repos!

jp-dbt commented:
> One change request, one nitpicky and up for negotiation request, and I think a changelog entry would be good for this one, but otherwise looks great!

Awesome, thanks! Addressed the comments, back over to you 😃

racheldaniel reviewed (approved):
🎉
